test/custom_task_test.go

```diff
@@ -552,6 +592,8 @@ func TestWaitCustomTask_PipelineRun(t *testing.T) {
 	for _, tc := range []struct {
 		name                      string
 		runDuration               string
+		runTimeout                string
+		runRetries                int
 		prTimeout                 string
 		prConditionAccessorFn     func(string) ConditionAccessorFn
 		wantPrConditionType       apis.ConditionType
```

Comment:
Instead of a bunch of different expected outputs, would it be possible to define a wantStatus that contains the expected custom run status?

Reply:
The `runStatus` differs upon different feature flags (FullEmbeddedStatus, BothEmbeddedStatus, MinimalEmbeddedStatus), therefore we need to compose it in runtime accordinlgy. wdyt?

Reply:
ok, in that case - shouldn't the feature flag value be one of the test inputs?

Reply:
Updated as suggested, use `wantRetriesStatus` to group related fields together.